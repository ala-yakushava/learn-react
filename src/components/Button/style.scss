@import "../../styles/theme";

.Button {
  @include reset-button;
  display: inline-block;
  padding: 12px 20px;
  font-size: 20px;
  text-transform: uppercase;
  border-radius: 5px;

  + .Button {
    margin-left: 15px;
  }

  &--primary {
    padding: 17px 75px;
    color: $white;
    background-color: $red;
  }

  &--secondary {
    color: $red;
    border: 2px solid $red;
    background-color: transparent;
  }

  &--transparent {
    color: $red;
    background-color: rgba($grey, 0.5);
  }

  &[disabled] {
    opacity: 0.5;
  }
}
