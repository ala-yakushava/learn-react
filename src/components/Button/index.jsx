import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import './style.scss';

const Button = ({ children, handleClick, type, className, mode, isDisabled }) => {
  const buttonClass = cn({
    Button: true,
    [`Button--${mode}`]: true,
    [className]: true,
  });

  return (
    <button type={type} disabled={isDisabled} className={buttonClass} onClick={handleClick}>
      <span className='Button_text'>
        {children}
      </span>
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.node,
  handleClick: PropTypes.func,
  type: PropTypes.oneOf(['button', 'submit', 'reset']),
  className: PropTypes.string,
  mode: PropTypes.string,
  isDisabled: PropTypes.bool,
};

Button.defaultProps = {
  handleClick: () => {},
  type: 'button',
  className: '',
  mode: 'default',
  isDisabled: false,
};

export default Button;
