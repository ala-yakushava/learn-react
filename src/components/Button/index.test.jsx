import React from 'react';
import renderer from 'react-test-renderer';

import Button from './index';

describe('Components - Button', () => {
  test('should match snapshot', () => {
    const tree = renderer
      .create(
        <Button>
          Click me!
        </Button>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
