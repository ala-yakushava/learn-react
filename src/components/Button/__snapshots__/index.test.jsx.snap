// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Button should match snapshot 1`] = `
<button
  className="Button Button--default"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <span
    className="Button_text"
  >
    Click me!
  </span>
</button>
`;
